src/components/TextInput/BaseTextInput.js

```diff
@@ -395,7 +395,7 @@ function BaseTextInput(props) {
                         setTextInputHeight(e.nativeEvent.layout.height);
                     }}
                 >
-                    {props.value || props.placeholder}
+                    {props.value ? `${props.value}\u200B` : props.placeholder}
```

Comment:
https://github.com/Expensify/App/pull/23286#issuecomment-1644174450

Reply:
@s-alves10 I saw the comment. 

> I've chosen \u200B(Zero-width space character) based on tests on various platforms.

Can you please explain. In which platform do we have the issue? And what issue do we have?

Reply:
Yes, I saw that adding \n increases the height in ios native. So I've done some research work about zero width character and found ZWSC(\u200B)

Reply:
```
Need to add \u200B(Zero-width space character) to get correct layout in cases value is ended with \n(EOL)
```
Is this comment ok?

Reply:
There is an issue on Android. An extra line is added if we enter any character.

https://github.com/Expensify/App/assets/25876548/177c820b-43bd-4a4f-9811-65d5b4142e46

Reply:
The other thing that I didn't understand is: Shouldn't we add the character only when user presses (i.e., line ends with `\n`)?

Reply:
https://github.com/Expensify/App/issues/21271#issuecomment-1643183002
I used alternative solution
Let me check again

Reply:
@sobitneupane 
I tested again but it work on all platforms. Can you check again?

The current change is based on the fact that ZWSC doesn't occupy space but it's still a character. \n character at the last position doesn't occupy space on some platforms, but if that is in the middle of the string, it will increase height on all platforms

Reply:
comment added

Reply:
> ```
> Need to add \u200B(Zero-width space character) to get correct layout in cases value is ended with \n(EOL)
> ```
> 
> Is this comment ok?

```\u200B added to solve the issue(https://github.com/Expensify/App/issues/21271): "Cursor not visible on hitting enter"```

Reply:
> The other thing that I didn't understand is: Shouldn't we add the character only when user presses (i.e., line ends with \n)?

I cannot find answer to this question in proposal as well. Can you please help me understand it?

Reply:
comment updated

Reply:
> I tested again but it works fine on all platforms. Can you check again?

I am bit surprised that I cannot reproduce it any longer.

>This adds the space character when the string ends with '\n' to prevent the string to be ended with '\n'. But we can solve this issue by simply adding ZWSC because this has a zero-width and assures that the string won't be ended with '\n'. At first, I suggested to use '\n', but it has a problem in native platforms, so I decided to use ZWSC

So, there is no solid reason to do so. Why don't we add zwsp character only if the value ends with `\n`.

Reply:
@s-alves10 Please let me know what you think about above comment.

Reply:
> Why don't we add zwsp character only if the value ends with `\n`

The issue happens when the value ends with `\n`.
You're right. We can solve this by adding extra ZWSC when the value ends with `\n`. But we can add ZWSC when the value doesn't end with `\n` as well because ZWSC has zero-width. The reason is for simplicity.

Reply:
@sobitneupane 

Should I update the code?

Reply:
> Should I update the code?

Yes, let's add the character only for necessary case.

Reply:
@sobitneupane 

Updated. Will you take a look again?

Reply:
@s-alves10 Can you please update the screenshots as well.

Reply:
Screenshots are the same as before. What's it for?

Reply:
> Screenshots are the same as before. I've verified that it works fine. Do I need to record them again?

@s-alves10 Yes.

Reply:
@sobitneupane 
Videos updated